elasticsearch/run.sh

```diff
@@ -87,6 +88,100 @@ sg_role_prometheus:
     - "${PROMETHEUS_USER:-system:serviceaccount:prometheus:prometheus}"
 CONF
 
+build_jks_from_pem() {
+
+  jks_name=$1
+  key_name=$2
+  cert_name=$3
+  ca_name=$4
+
+  openssl                                   \
+    pkcs12                                  \
+    -export                                 \
+    -in $secret_dir/$cert_name              \
+    -inkey $secret_dir/$key_name            \
+    -out $secret_dir/$jks_name.p12          \
+    -passout pass:kspass
+
+  keytool                                   \
+    -importkeystore                         \
+    -srckeystore $secret_dir/$jks_name.p12  \
+    -srcstoretype PKCS12                    \
+    -srcstorepass kspass                    \
+    -destkeystore $secret_dir/$jks_name.jks \
+    -deststoretype JKS                      \
+    -deststorepass kspass                   \
+    -noprompt
+
+  keytool                                   \
+    -changealias                            \
+    -keystore $secret_dir/$jks_name.jks     \
+    -storepass kspass                       \
+    -alias 1                                \
+    -destalias $jks_name
+
+  keytool                                   \
+    -import                                 \
+    -file $secret_dir/$ca_name              \
+    -keystore $secret_dir/$jks_name.jks     \
+    -storepass kspass                       \
+    -noprompt                               \
+    -alias sig-ca
+}
+
+copy_keys_to_secretdir() {
+
+  if [ -d $provided_secret_dir ] ; then
+    info "Copying certs from ${provided_secret_dir} to ${secret_dir}"
+    cp $provided_secret_dir/* $secret_dir/
+  fi
+}
+
+# Pull in the certs provided in our secret and generate our necessary jks and truststore files
+build_jks_truststores() {
+
+  copy_keys_to_secretdir
+
+  info "Building required jks files and truststore"
+
+  # check for lack of admin.jks
+  if [[ ! -e $secret_dir/admin.jks ]]; then
+    build_jks_from_pem "admin" "admin-key" "admin-cert" "admin-ca"
+  fi
+
+  # check for elasticsearch.key and elasticsearch.crt
+  if [[ -e $secret_dir/elasticsearch.key && -e $secret_dir/elasticsearch.crt ]]; then
```

Comment:
you check for the existence of `admin.jks` before writing it - do you need to check for `elasticsearch.jks` and `logging-es.jks` before writing them?

Reply:
so my thought here was if we have the .key and .crt then that would mean we shouldn't have the jks as generated by ansible. There shouldn't be any harm in adding the jks check though... I'll do that.